### DISCUSSION TITLE
rpathdirs  onnxruntime

### DISCUSSION THREAD

**Discussion by youngday** [xmake-io/xmake#3393] [Q&A]
大家好，
1，使用onnxruntime时，xmake 编译调试都没问题，在终端运行时，需要拷贝.so文件才可以运行;
2，使用本地库onnxruntime-gpu 版本时，编译没问题，调试找不到.so,报错：
```sh
 error while loading shared libraries: libonnxruntime.so.1.12.1: cannot open shared object file: No such file or directory
```
在终端运行，拷贝.so，运行正常，路径确认多次，env也根据rpathdir LD_LIBRARY_PATH设置过，没有成功。
3，如何在linux下自动打包，查找.so库等文件呢？xmake 有类似 conan 的 deploy功能吗？看到只有android。
感谢大家。

**Comment by waruqi** [comment]
走  xmake install -o outputdir 安装，或者直接 xmake run 运行

**Reply by youngday** [reply]
直接xmake run 可以了，现在调试也可以了，都正常了。不过同时我也升级gcc到gcc11了，准备c++20了，感谢您
